sdk/contentsafety/ai-content-safety-rest/test/public/contentSafety.spec.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Recorder } from "@azure-tools/test-recorder";
+import { assert } from "chai";
+import { createRecorder, createClient } from "./utils/recordedClient";
+import { Context } from "mocha";
+import { ContentSafetyClient, isUnexpected } from "../../src";
+import fs from "fs";
+import path from "path";
+import { isBrowser } from "@azure/core-util";
+
+describe("Content Safety Client Test", () => {
+  let recorder: Recorder;
+  let client: ContentSafetyClient;
+  function sleep(time: number): Promise<NodeJS.Timer> {
+    return new Promise((resolve) => setTimeout(resolve, time));
+  }
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    client = await createClient(recorder);
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("analyze text", async function () {
+    const response = await client.path("/text:analyze").post({
+      body: {
+        text: "You are an idiot",
+        categories: ["Hate"],
+      },
+    });
+    if (isUnexpected(response)) {
+      throw new Error(response.body?.error.message);
+    }
+    assert.strictEqual(response.status, "200");
+    assert.equal(response.body.hateResult?.category, "Hate");
+    assert.notExists(response.body.selfHarmResult);
+  });
+
+  it("analyze image", async function () {
+    let base64Image: string;
+    if (isBrowser) {
+      const imagePath = "http://localhost:9876/base/samples-dev/example-data/image.png";
+      const response = await fetch(imagePath);
+      const buffer = await response.arrayBuffer();
+      const binary = new Uint8Array(buffer);
+      const chunkSize = 0x8000;
+      const chunks = [];
+      for (let i = 0; i < binary.length; i += chunkSize) {
+        chunks.push(binary.slice(i, i + chunkSize));
+      }
+      const stringChunks = chunks.map((chunk) =>
+        String.fromCharCode.apply(null, Array.from(chunk))
+      );
+      base64Image = btoa(stringChunks.join(""));
+    } else {
+      const imagePath = path.join("samples-dev", "example-data", "image.png");
+      const buffer = fs.readFileSync(imagePath);
```

Comment:
I wonder if we should add some helpers in core-util for function with similiar functionalities could be work in both Node and Browser environment ? kind of hit issues like this a lot. @joheredi @xirzec